File: src/socket/SocketType.cpp
//
// Created by Andrei on 15-Apr-21.
//

#include <comm/socket/SocketType.h>
#include <stdexcept>

using namespace comm;
using namespace std;

SocketType comm::stringToSocketType(const string &s) {
    if (s == "UDP") {
        return SocketType::UDP;
    } else if (s == "UDP_HEADER") {
        return SocketType::UDP_HEADER;
    } else if (s == "TCP") {
        return SocketType::TCP;
    } else {
        throw runtime_error("Unknown SocketType " + s);
    }
}

string comm::socketTypeToString(SocketType socketType) {
    switch (socketType) {
        case UDP: {
            return "UDP";
        }
        case UDP_HEADER: {
            return "UDP_HEADER";
        }
        case TCP: {
            return "TCP";
        }
        default : {
            throw runtime_error("Unknown SocketType " + to_string(socketType));
        }
    }
}

File: csharp/socket/SocketType.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DesignAIRobotics.Comm.socket {
    public enum SocketType {
        UDP,
        UDP_HEADER,
        TCP,
    }

    public static class SocketTypeConverter {
        public static SocketType stringToSocketType(string s) {
            if (s == "UDP") {
                return SocketType.UDP;
            } else if (s == "UDP_HEADER") {
                return SocketType.UDP_HEADER;
            } else if (s == "TCP") {
                return SocketType.TCP;
            } else {
                throw new Exception("Unknown SocketType " + s);
            }
        }

        public static string socketTypeToString(SocketType socketType) {
            switch (socketType) {
                case SocketType.UDP: {
                    return "UDP";
                }
                case SocketType.UDP_HEADER: {
                    return "UDP_HEADER";
                }
                case SocketType.TCP: {
                    return "TCP";
                }
                default: {
                    throw new Exception("Unknown SocketType " + socketType);
                }
            }
        }
    }
}
